Source/WebKit/WebProcess/WebProcess.cpp

```diff
@@ -1677,12 +1677,31 @@ void WebProcess::nonVisibleProcessEarlyMemoryCleanupTimerFired()
     if (!m_pagesInWindows.isEmpty())
         return;
 
+    destroyDecodedDataForAllImages();
+
 #if PLATFORM(COCOA)
     destroyRenderingResources();
     releaseSystemMallocMemory();
 #endif
 }
 
+void WebProcess::destroyDecodedDataForAllImages()
+{
+    for (auto& page : m_pageMap.values()) {
+        page->releaseAllRemoteImageResources();
+        if (auto* drawingArea = page->drawingArea())
+            drawingArea->setNextRenderingUpdateRequiresSynchronousImageDecoding();
```

Comment:
Can this be part of WebPage::releaseAllRemoteImageResources()? The WebPage is destroying the decoded data so it makes sense to mark its DrawingArea for synchronous image decoding.

Reply:
It's really the destroyDecodedDataForAllImages() call in WebProcess::destroyDecodedDataForAllImages that is the thing that needs this flag set, not the remote resource releasing (which would be fine even without setting the flag). So I think it makes more sense here.

Reply:
I guess it could go either way; the destroyDecodedDataForAllImages works on all pages' image resources (since it's the process global MemoryCache it's working on).